AckooSDK/Classes/Networking/AckooError.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  AckooError.swift
+//  AckooSDK
+//
+//  Created by Sally Ahmed1 on 11/23/20.
+//
+
+import Foundation
+struct BackEndError :Codable{
+    let ok: Bool
+    let error: AckooError
+}
+
+protocol AckooErrorType: Error {
+    var message: String { get }
+    var code: String { get }
+}
+public struct AckooError: AckooErrorType, Codable {
+    var code: String
+    var message: String
+    private(set) var api: String?
```

Comment:
do we add details here ? 
remember that for validation errors the details of the errors are sent in details ? 
I think we may need to dicuss this further